dart/lib/src/sentry.dart

```diff
@@ -45,13 +45,14 @@ class Sentry {
   }) async {
     final sentryOptions = options ?? SentryOptions();
 
-    await _initDefaultValues(sentryOptions);
+    _setEnvironmentVariables(sentryOptions);
 
     try {
       final config = optionsConfiguration(sentryOptions);
       if (config is Future) {
         await config;
       }
+      await _initDefaultValues(sentryOptions);
```

Comment:
Changing this order probably breaks stuff in unpredictable ways.

Reply:
Why do we have to change the ordering?

Reply:
Ok nvm I understand. 

We'll have to think further about this. Like you said this might break stuff / have side effects

Reply:
We can have a static `setWindow()` function which sets an internal window and then assign that to options when calling init.

```dart
Window _window = NoopWindow();
void setWindow(Window window) {
  if (!PlatformChecker().isWeb) {
    return;
  }
  _window = window;
}

// and then inside init()
final sentryOptions = options ?? SentryOptions();

options?.window = _window;
```

The user just needs to call the setWindow function before SentryFlutter.init

```dart
Sentry.setWindow(...)
SentryFlutter.init(...)
```

at least that way we can keep the ordering. maybe there's better ways

Reply:
This is still a breaking change, isn't it? As in, things that worked before for users won't work anymore unless they change their code.

Reply:
> This is still a breaking change, isn't it? As in, things that worked before for users won't work anymore unless they change their code.

tbh I'm not sure, wasm compilation only works on the latest stable channel and it's not working right now for them anyway.

non-wasm users won't have to care about this stuff

Reply:
What I meant is that it would be breaking for existing users targeting web? Or am I missing something.